### PR TITLE
Add equals and hashValue methods for each dto class

### PR DESCRIPTION
**Pull request by KochTobi**
The DTOs did not provide `equals` and `hashValue` methods. This PR adds those methods with the assumption that all fields can be null since the classes in question are DTOs and only provide the data fields.

### CODE REVIEW

File: src/main/groovy/life/qbic/datamodel/dtos/business/Affiliation.groovy [hunk 1]
```diff
@@ -42,7 +42,35 @@ final class Affiliation {
...
+    if (organisation != that.organisation) return false
+    if (postalCode != that.postalCode) return false
+    if (street != that.street) return false
+
+    return true
+  }
+
+  int hashCode() {
+    int result
+    result = (organisation != null ? organisation.hashCode() : 0)
```
Comment (jenniferboedker) [inline src/main/groovy/life/qbic/datamodel/dtos/business/Affiliation.groovy:64]: why not 
```suggestion
    result = 31* result + (organisation != null ? organisation.hashCode() : 0)
```

File: src/main/groovy/life/qbic/datamodel/dtos/imaging/Image.groovy [hunk 2]
```diff
@@ -101,4 +101,37 @@ final class Image {
...
+    if (modality != image.modality) return false
+    if (pixelSize != image.pixelSize) return false
+    if (resolution != image.resolution) return false
+
+    return true
+  }
+
+  int hashCode() {
+    int result
+    result = (acquisitionDate != null ? acquisitionDate.hashCode() : 0)
```
Comment (sven1103) [inline src/main/groovy/life/qbic/datamodel/dtos/imaging/Image.groovy:126]: Elvis ftw
```suggestion
    result = ( acquisitionDate?.hashCode() ?: 0)
```

**Review by sven1103 — CHANGES_REQUESTED** [review]
Please use the elvis operator and null safe operator, that is syntactic sugar of Groovy to avoid null checks.

And here, its application makes a lot of sense and improves readability of the code.

File: src/main/groovy/life/qbic/datamodel/dtos/imaging/Image.groovy (hunk 2 above)
Comment (KochTobi) [reply src/main/groovy/life/qbic/datamodel/dtos/imaging/Image.groovy:126]: I will do that in a separate commit. The solution above is the default intellij groovy hashCode method.

**Comment by KochTobi** [comment]
> Please use the elvis operator and null safe operator, that is syntactic sugar of Groovy to avoid null checks.
> 
> And here, its application makes a lot of sense and improves readability of the code.

Do you think we could also use 
```
 import groovy.transform.EqualsAndHashCode
 @EqualsAndHashCode
```

**Review by sven1103 — APPROVED** [review]
400 less lines of code, i like :)

**Pull request merged**